### PR TITLE
TSK-1946: update to Spring Boot 2.7.3

### PR DESCRIPTION
**Pull request by norman-schmidt**
---
Release Notes:

```

```
 
### For the submitter:
- [ ] I updated the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) and will supply links to the specific files
- [x] I did not update the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview)
- [x] I included a link to the [SonarCloud branch analysis](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1019969636/SonarCloud+Integration)
- https://sonarcloud.io/summary/new_code?id=norman-schmidt_taskana&branch=spring-update
Quality Gate fails but nothing really has changed, duplicates rather have been removed (security configs) 
- [ ] After integration of the PR, I added a description of changes on the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I did not update the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [ ] I put the ticket in review
- [ ] After integration of the pull request, I verified our [bluemix test environment](http://taskana.mybluemix.net/taskana) is not broken

### Verified by the reviewer:
- [x] Commit message format → TSK-XXX: Your commit message.
- [x] Submitter's update to [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) is sufficient
- [x] SonarCloud analysis meets our standards
- [ ] Update of the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana) reflects changes
- [ ] PR fulfills the ticket
- [ ] Edge cases and unwanted side effects are tested
- [ ] Readability

### CODE REVIEW

File: history/taskana-simplehistory-rest-spring/src/test/java/pro/taskana/simplehistory/rest/HistoryWebSecurityConfigurer.java [hunk 1]
```diff
@@ -2,62 +2,66 @@
...
   private final String ldapServerUrl;
   private final String ldapBaseDn;
-  private final String ldapGroupSearchBase;
   private final String ldapUserDnPatterns;
 
   @Autowired
   public HistoryWebSecurityConfigurer(
       @Value("${taskana.ldap.serverUrl:ldap://localhost:10389}") String ldapServerUrl,
       @Value("${taskana.ldap.baseDn:OU=Test,O=TASKANA}") String ldapBaseDn,
-      @Value("${taskana.ldap.groupSearchBase:cn=groups}") String ldapGroupSearchBase,
```
Comment (mustaphazorgati) [inline history/taskana-simplehistory-rest-spring/src/test/java/pro/taskana/simplehistory/rest/HistoryWebSecurityConfigurer.java:33]: What happened to this property? Where is it now?

File: rest/taskana-rest-spring-example-wildfly/src/test/java/pro/taskana/example/wildfly/TaskanaWildflyWithHistoryLoggerEnabledTest.java [hunk 2]
```diff
@@ -1,3 +1,4 @@
+/*
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring-example-wildfly/src/test/java/pro/taskana/example/wildfly/TaskanaWildflyWithHistoryLoggerEnabledTest.java]: unnecessary diff. Please fix

File: rest/taskana-rest-spring-example-wildfly/src/test/java/pro/taskana/example/wildfly/TaskanaWildflyWithSimpleHistoryAndHistoryLoggerEnabledTest.java [hunk 3]
```diff
@@ -1,3 +1,4 @@
+/*
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring-example-wildfly/src/test/java/pro/taskana/example/wildfly/TaskanaWildflyWithSimpleHistoryAndHistoryLoggerEnabledTest.java]: unnecessary diff. Please fix

File: rest/taskana-rest-spring-example-wildfly/src/test/java/pro/taskana/example/wildfly/TaskanaWildflyWithSimpleHistoryEnabledTest.java [hunk 4]
```diff
@@ -1,3 +1,4 @@
+/*
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring-example-wildfly/src/test/java/pro/taskana/example/wildfly/TaskanaWildflyWithSimpleHistoryEnabledTest.java]: Unnecessary diff. Please fix

File: routing/taskana-routing-rest/target\routing.dmn [hunk 5]
```diff
@@ -0,0 +1,63 @@
+<?xml version="1.0" encoding="UTF-8"?><definitions xmlns="http://www.omg.org/spec/DMN/20151101/dmn.xsd" expressionLanguage="http://www.omg.org/spec/FEEL/20140401" id="definitions" name="definitions" namespace="http://camunda.org/schema/1.0/dmn" typeLanguage="http://www.omg.org/spec/FEEL/20140401">
```
Comment (mustaphazorgati) [inline routing/taskana-routing-rest/target\routing.dmn:1]: remove

**Review by mustaphazorgati — CHANGES_REQUESTED** [review]

File: rest/taskana-rest-spring-example-common/src/test/java/pro/taskana/example/CommonWebSecurityConfigurer.java [hunk 6]
```diff
@@ -2,59 +2,63 @@
...
+    return new DefaultSpringSecurityContextSource(ldapServerUrl + "/" + ldapBaseDn);
+  }
+
+  @Bean
+  AuthenticationManager ldapAuthenticationManager(
+      BaseLdapPathContextSource contextSource, LdapAuthoritiesPopulator authorities) {
+    @SuppressWarnings("deprecation")
+    LdapPasswordComparisonAuthenticationManagerFactory factory =
+        new LdapPasswordComparisonAuthenticationManagerFactory(
+            contextSource, NoOpPasswordEncoder.getInstance());
```
[the remark this answers is not in the document]
Comment (norman-schmidt) [reply rest/taskana-rest-spring-example-common/src/test/java/pro/taskana/example/CommonWebSecurityConfigurer.java]: I'd say yes in case we want to keep it like that. In case we want to use Password Encoding in the future we should raise a ticket for that.

**Comment by norman-schmidt** [comment]
Wie in den Diagrammen oben zu sehen, hab ich die Beans aus den src-packages (grün) in rest-spring-example-common zu rest-spring-example-boot gezogen, weil die nicht vom Wildfly example genutzt werden, so wie ichs verstanden/getestet habe und dementsprechend auch nicht "common" sind.

File: history/taskana-simplehistory-rest-spring/src/test/java/pro/taskana/simplehistory/rest/HistoryWebSecurityConfigurer.java (hunk 1 above)
Comment (mustaphazorgati) [reply history/taskana-simplehistory-rest-spring/src/test/java/pro/taskana/simplehistory/rest/HistoryWebSecurityConfigurer.java:33]: -> authoritiespopulator

File: rest/taskana-rest-spring-example-common/src/test/java/pro/taskana/example/CommonWebSecurityConfigurer.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (mustaphazorgati) [reply rest/taskana-rest-spring-example-common/src/test/java/pro/taskana/example/CommonWebSecurityConfigurer.java]: Where's the comment? :D

File: rest/taskana-rest-spring-test/pom.xml [hunk 7]
```diff
@@ -0,0 +1,110 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <artifactId>taskana-rest-spring-test</artifactId>
+
+  <name>${project.groupId}:${project.artifactId}</name>
+  <description>Test utils for the taskana rest logic.</description>
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring-test/pom.xml:9]: Test utils for TASKANA rest logic (based on spring)

File: rest/taskana-rest-spring-test/pom.xml [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+      <artifactId>junit-jupiter-api</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.springframework.restdocs</groupId>
+      <artifactId>spring-restdocs-core</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>capital.scalable</groupId>
+      <artifactId>spring-auto-restdocs-core</artifactId>
+      <version>2.0.11</version>
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring-test/pom.xml:102]: please refer to parent pom properties

File: rest/taskana-rest-spring-test/pom.xml [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+    <version>5.6.1-SNAPSHOT</version>
+    <relativePath>../pom.xml</relativePath>
+  </parent>
+
+  <properties>
+    <sonar.tests/>
+  </properties>
+
+  <dependencies>
+    <dependency>
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring-test/pom.xml:23]: please try to sort and group the dependencies.
Example: all taskana (internal) modules should stay together.

File: rest/taskana-rest-spring-test/pom.xml [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+    </dependency>
+    <dependency>
+      <groupId>org.springframework.restdocs</groupId>
+      <artifactId>spring-restdocs-core</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>capital.scalable</groupId>
+      <artifactId>spring-auto-restdocs-core</artifactId>
+      <version>2.0.11</version>
+      <scope>compile</scope>
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring-test/pom.xml:103]: Is there a reason you modified the scope?

File: rest/taskana-rest-spring-test/src/main/java/pro/taskana/rest/test/SpringSecurityToJaasFilter.java [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.web.filter.GenericFilterBean;
+
+import pro.taskana.common.api.exceptions.SystemException;
+import pro.taskana.common.api.security.GroupPrincipal;
+import pro.taskana.common.api.security.UserPrincipal;
+
+/** Simple Filter to map all Spring Security Roles to JAAS-Principals. */
+public class SpringSecurityToJaasFilter extends GenericFilterBean {
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring-test/src/main/java/pro/taskana/rest/test/SpringSecurityToJaasFilter.java:23]: Please mark this class as a duplicate

File: rest/taskana-rest-spring-test/src/main/java/pro/taskana/rest/test/TestWebSecurityConfig.java [hunk 12]
```diff
@@ -77,6 +88,7 @@ public LdapAuthoritiesPopulator authoritiesPopulator(
   }
 
   @Bean
+  @Primary
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring-test/src/main/java/pro/taskana/rest/test/TestWebSecurityConfig.java:91]: Why?

File: rest/taskana-rest-spring-test/src/main/resources/taskana-test.ldif [hunk 13]
```diff
@@ -0,0 +1,449 @@
+######################
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring-test/src/main/resources/taskana-test.ldif:1]: Please delete old `taskana-test.ldif` file

File: rest/taskana-rest-spring/pom.xml [hunk 14]
```diff
@@ -170,6 +170,12 @@
       <version>${version.archunit}</version>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>pro.taskana</groupId>
+      <artifactId>taskana-test-api</artifactId>
+      <version>${project.version}</version>
+      <scope>test</scope>
+    </dependency>
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/pom.xml:178]: For what do you use the test-api?

File: routing/taskana-routing-rest/pom.xml [hunk 15]
```diff
@@ -141,6 +141,18 @@
...
+      <artifactId>taskana-test-api</artifactId>
+      <version>${project.version}</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>pro.taskana</groupId>
+      <artifactId>taskana-common-test</artifactId>
+      <version>${project.version}</version>
+      <scope>test</scope>
+    </dependency>
```
Comment (mustaphazorgati) [inline routing/taskana-routing-rest/pom.xml:155]: group taskana-dependencies together.

Why do we need these dependencies?

**Review by mustaphazorgati — CHANGES_REQUESTED** [review]
* You forgot to move the asciidoc templates from taskana-common-test to taskana-rest-spring-test
* We should find a better name. `taskana-rest-spring-test` could also mean that it contains all tests for the `taskana-rest-spring` module.

File: rest/taskana-rest-spring-test/pom.xml (hunk 8 above)
Comment (norman-schmidt) [reply rest/taskana-rest-spring-test/pom.xml:102]: What do you mean?

File: rest/taskana-rest-spring-test/src/main/java/pro/taskana/rest/test/TestWebSecurityConfig.java (hunk 12 above)
Comment (norman-schmidt) [reply rest/taskana-rest-spring-test/src/main/java/pro/taskana/rest/test/TestWebSecurityConfig.java:91]: Because `rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/ldap/LdapConfiguration.java -> ldapContextSource()` also creates a bean of type `org.springframework.ldap.core.support.BaseLdapPathContextSource`, and for the tests we want the one from the TestWebSecurityConfig.

File: rest/taskana-rest-spring/pom.xml (hunk 14 above)
Comment (norman-schmidt) [reply rest/taskana-rest-spring/pom.xml:178]: For these annotations:
import pro.taskana.testapi.security.JaasExtension;
import pro.taskana.testapi.security.WithAccessId;

Previously we got them from taskana-common-test, from which we detached by creating the new module.

File: routing/taskana-routing-rest/pom.xml (hunk 15 above)
Comment (norman-schmidt) [reply routing/taskana-routing-rest/pom.xml:155]: Same here test-api is used for the annotations, although routing-rest still depends on other common-test classes, which means we can still use the two annotations from common-test. I will remove the test-api dependency

File: rest/taskana-rest-spring-test/src/main/java/pro/taskana/rest/test/SpringSecurityToJaasFilter.java (hunk 11 above)
Comment (norman-schmidt) [reply rest/taskana-rest-spring-test/src/main/java/pro/taskana/rest/test/SpringSecurityToJaasFilter.java:23]: Like this?

/**
 * Simple Filter to map all Spring Security Roles to JAAS-Principals.
 *
 * <p>Duplicate of {@linkplain pro.taskana.rest.test.SpringSecurityToJaasFilter}
 */

Comment (norman-schmidt) [reply rest/taskana-rest-spring-test/src/main/java/pro/taskana/rest/test/SpringSecurityToJaasFilter.java:23]: Like this?

/**
 * Simple Filter to map all Spring Security Roles to JAAS-Principals.
 *
 * <p>Duplicate of {@linkplain pro.taskana.common.rest.SpringSecurityToJaasFilter}
 */

**Review by arolfes — APPROVED** [review]
Looks good to me
